src/box/replication.cc

```diff
@@ -1251,7 +1260,13 @@ replicaset_connect(const struct uri_set *uris,
 	memset(&state, 0, sizeof(state));
 	fiber_cond_create(&state.wakeup);
 
-	if (uris->uri_count == 0) {
+	/*
+	 * The supervised bootstrap strategy has an exception
+	 * here, because it waits not only for replicas, but also
+	 * for local bootstrap leadership flags.
+	 */
+	if (bootstrap_strategy != BOOTSTRAP_STRATEGY_SUPERVISED &&
```

Comment:
A meaningful log msg would be great here. This is not very self explanatory:

```
2025-04-09 19:29:06.848 [378076] main/104/interactive box.cc:5573 I> instance uuid 28e82fd7-34f9-41f7-8595-710acc2ed917
2025-04-09 19:29:06.848 [378076] main/104/interactive replication.cc:1294 I> connecting to 0 replicas
```

Reply:
Good point, thanks!

Done this way:

```diff
diff --git a/src/box/replication.cc b/src/box/replication.cc
index 347c0b62de..d74a3a028f 100644
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -1291,7 +1291,11 @@ replicaset_connect(const struct uri_set *uris,
 	for (; count < uris->uri_count; count++)
 		appliers[count] = applier_new(&uris->uris[count]);
 
-	say_info("connecting to %d replicas", count);
+	if (count == 0)
+		say_info("no upstreams are configured, waiting for the "
+			 "bootstrap command");
+	else
+		say_info("connecting to %d replicas", count);
 
 	if (!connect_quorum) {
 		/*
```